src/Shared/FileUtilities.cs

```diff
@@ -1078,7 +1078,12 @@ internal static string MakeRelative(string basePath, string path)
             {
                 sb.Append(splitPath[i]).Append(Path.DirectorySeparatorChar);
             }
-            sb.Length--;
+
+            if (fullPath[fullPath.Length - 1] != Path.DirectorySeparatorChar)
```

Comment:
should it also check Path.AlternateDirectorySeparatorChar?

Reply:
My understanding is that the `fullPath` has already been normalised within the `Path.GetFullPath` function so should only contain `Path.DirectorySeparatorChar`?

Reply:
Looks like that's true for [unix](https://github.com/dotnet/runtime/blob/8709d595f7105cbf73774ec762266bc56f99cf04/src/libraries/System.Private.CoreLib/src/System/IO/PathInternal.cs#L196-L200). [Windows](https://github.com/dotnet/runtime/blob/8709d595f7105cbf73774ec762266bc56f99cf04/src/libraries/Common/src/Interop/Windows/Kernel32/Interop.GetFullPathNameW.cs#L15) calls down into the kernel, but I'm assuming it's the same.

Reply:
Sure is:

```csharp
Microsoft (R) Visual C# Interactive Compiler version 3.10.0-4.21269.26 ()
Loading context from 'CSharpInteractive.rsp'.
Type "#help" for more information.
> System.Console.WriteLine(System.IO.Path.GetFullPath("C:/abc/def/"));
C:\abc\def\
> System.Console.WriteLine(System.IO.Path.GetFullPath("C:/abc/def"));
C:\abc\def
> 
```